### PR TITLE
Refactored CreateUser to use a handler function

### PR DESCRIPTION
**Pull request by samarthupadhyaya27**

### CODE REVIEW

File: handler/user_test.go [hunk 1]
```diff
@@ -218,3 +218,26 @@ func TestDeleteUser(t *testing.T) {
 		}
 	})
 }
+
+func TestCreateUser(t *testing.T) {
```
Comment (lumisphere902) [inline handler/user_test.go:222]: Test cases that fall under CreateUser should be in this format:
```
t.Run("test case", func(t *testing.T) {
...
}
```
You don't need to fully test every case right now, but it would be great if you could format it and do the tests in a separate PR. :)

**Review by lumisphere902 — CHANGES_REQUESTED** [review]
Hey Samarth, sorry for the late response. This looks pretty good overall; there are just a few comments I left about the tests. Keep up the good work!

File: db/mock_test.go [hunk 2]
```diff
@@ -76,6 +86,10 @@ func TestMockClass(t *testing.T) {
 		_, err := d.LoadClass(context.Background(), "test")
 		assert.NoError(t, err)
 	})
+	t.Run("create", func(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (samarthupadhyaya27) [reply db/mock_test.go:89]: I don't remember adding that, I must have made a mistake. I only meant to add tests for CreateUser and CreateProgram which are handler functions I created. I'll remove this extra test. Sorry about that.

File: handler/user_test.go (hunk 1 above)
Comment (samarthupadhyaya27) [reply handler/user_test.go:222]: I was a little confused about this since the original tests for CreateUser were the only tests in db/user.go that weren't written in the form above. I'll change the tests to be written in that format. Thanks for your review.

**Review by lumisphere902 — APPROVED** [review]
LGTM. Thanks samarth, and congrats on your first Go PR! Feel free to squash and merge :)
